### ISSUE TITLE
set_auto_maskandscale and valid_min/max behavior when writing

### ISSUE THREAD

**Issue by jdemaria** [Unidata/netcdf4-python#1094]
Hi,

From the docs:
https://unidata.github.io/netcdf4-python/#Variable.set_auto_maskandscale
"When data is written to a variable, the masked array is converted back to a regular numpy array by replacing all the masked values by the missing_value attribute of the variable (if it exists). If the variable has no missing_value attribute, the _FillValue is used instead. **If the variable has valid_min/valid_max and missing_value attributes, data outside the specified range will be set to missing_value.**"

But the code below does not set values outside valid_min/max to missing_value. Is it normal?
For me the current behavior seems logical: in general when writing we want not that the values outside valid_min/max will be automatically set to missing_value (but when reading we want). So maybe it's just a bug in the documentation?

My config:
Ubuntu 18.04.5 LTS
netcdf4                   1.5.6           nompi_py38h1cdf482_100    conda-forge

Thanks in advance,
Julien

```
import netCDF4 as nc
import numpy as np

d = nc.Dataset('bug.nc', 'w')
d.createDimension('dim', 5)
v = d.createVariable('v', 'f4', dimensions=('dim',), fill_value=-999)
v.set_auto_maskandscale(True)
v.missing_value = v._FillValue
v.valid_min = 2
v.valid_max = 3
ma = np.ma.masked_equal([0., 1, 2, 3, 4], 0)
print(ma)
v[:] = ma
d.close()

d = nc.Dataset('bug.nc')
v = d.variables['v']
v.set_auto_maskandscale(False)
print(v[:])
d.close()
```

**Comment by jswhit** [comment]
This is a doc bug. valid_range is not checked for writes, only reads.

**Comment by jswhit** [comment]
closed by #1095
